packages/core/src/scripts/migrate-legacy-project.ts

```diff
@@ -0,0 +1,251 @@
+import path from 'path';
+import { promises as fs } from 'fs';
+import type { ManifestData, ImplDeployment } from '../manifest';
+import type { StorageItem, StorageLayout, TypeItem } from '../storage';
+
+const OPEN_ZEPPELIN_FOLDER = '.openzeppelin';
+const MIGRATION_FILE_LOCATION = 'openzeppelin-cli-export.json';
+
+export async function migrateLegacyProject(): Promise<void> {
+  const oldManifests = await getManifests();
+  const { newManifests, exportData } = await migrateManifestFiles(oldManifests);
+
+  for (const manifestFile in newManifests) {
+    const newManifest = newManifests[manifestFile];
+    await writeJSONFile(getNewManifestLocation(manifestFile), newManifest);
+  }
+
+  await writeJSONFile(MIGRATION_FILE_LOCATION, exportData);
+}
+
+async function migrateManifestFiles(oldManifests: string[]): Promise<MigrationOutput> {
+  const exportData: Record<string, unknown> = {};
+  const newManifests: Record<string, ManifestData> = {};
+
+  for (const manifestFile of oldManifests) {
+    const oldManifest = JSON.parse(await fs.readFile(manifestFile, 'utf8'));
+
+    if (oldManifest.manifestVersion === undefined) {
+      throw new Error('Migration failed: manifest version too old');
+    }
+
+    const [major, minor] = oldManifest.manifestVersion.split('.');
+
+    if (major === '3') {
+      continue;
+    }
+
+    // 2.2 latest version
+    if (major !== '2' || minor !== '2') {
+      throw new Error(`Migration failed: expected manifest version 2.2, got ${major}.${minor} instead`);
```

Comment:
Maybe we could point them somewhere to update it, otherwise this is a deadend.

Reply:
I couldn't find documentation for this. The recommendation should be to first update to the latest version of OpenZeppelin CLI, then migrate to the plugins.